File: QytetetRuby/lib/sorpresa.rb
#encoding:utf-8
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.

module ModeloQytetet
  class Sorpresa
    def initialize(unTexto, unValor, unTipo)
      @texto = unTexto #testo de carta Sorpresa
      @valor = unValor #valor de sorpresa
      @tipo = unTipo #tipo de sorpresa de TipoSorpresa
    end
    
    protected
    attr_reader:texto, :valor, :tipo

    public
    def to_s()
      "Texto: #{@texto} \n Valor: #{@valor} \n Tipo: #{@tipo}"
    end

  end
end

File: QytetetJava/src/modeloqytetet/Sorpresa.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package modeloqytetet;

/**
 *
 * @author venrra
 */
public class Sorpresa {
    private String texto;
    private TipoSorpresa tipo;
    private int valor;
    
    Sorpresa(String texto, int valor, TipoSorpresa tipo){
        this.texto = texto;
        this.tipo = tipo;
        this.valor = valor;
    }
    
    String getTexto(){
        return this.texto;
    }
    
    TipoSorpresa getTipo(){
        return this.tipo;
    }
    
    int getValor(){
        return this.valor;
    }
    
    @Override
    public String toString(){
        return "Sorpresa{" + "texto=" + texto + ", valor= " + Integer.toString(valor) + ", tipo=" + tipo +"}";
    }
}
